subsys/mgmt/mcumgr/Kconfig

```diff
@@ -6,6 +6,7 @@ menuconfig MCUMGR
 	bool "mcumgr Support"
 	select NET_BUF
 	select ZCBOR
+	select CRC
```

Comment:
Move to:

`subsys/mgmt/mcumgr/transport/Kconfig.dummy` in `MCUMGR_SMP_DUMMY`
`subsys/mgmt/mcumgr/transport/Kconfig.shell` in `MCUMGR_SMP_SHELL`
`subsys/mgmt/mcumgr/transport/Kconfig.uart` in `MCUMGR_SMP_UART`
`subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/Kconfig` in `FS_MGMT_CHECKSUM_IEEE_CRC32`